typescript/sdk/src/token/deploy.ts

```diff
@@ -594,6 +600,54 @@ export class HypERC20Deployer extends TokenDeployer<HypERC20Factories> {
       implementationAddress,
     );
   }
+
+  async deployAndConfigureTokenFees(
+    deployedContractsMap: HyperlaneContractsMap<HypERC20Factories>,
+    configMap: ChainMap<HypTokenRouterConfig>,
+  ): Promise<void> {
+    await Promise.all(
+      Object.keys(deployedContractsMap).map(async (chain) => {
+        const config = configMap[chain];
+        const tokenFeeInput = config?.tokenFee;
+        if (!tokenFeeInput) return;
+
+        if (this.multiProvider.getProtocol(chain) !== ProtocolType.Ethereum) {
+          this.logger.debug(`Skipping token fee on non-EVM chain ${chain}`);
+          return;
+        }
+
+        if (!tokenFeeInput.owner) {
+          this.logger.debug(
+            `Token fee owner not specified, using router owner ${config.owner}`,
+          );
+          tokenFeeInput.owner = config.owner;
+        }
+
+        if (isCollateralTokenConfig(config) && !tokenFeeInput.token) {
+          this.logger.debug(
+            `Token fee token not specified for ${config.type}, using router token ${config.token}`,
+          );
+          tokenFeeInput.token = config.token;
+        }
```

Comment:
Should we maybe enforce at warp deploy config level that the tokens match? or are we planning to have different token types to pay the fees?

Reply:
Also wondering if we should enforce at the warp deploy config level that the destinations of the token fee router are the connected chains of the warp route or a subset

Reply:
good idea. I'm going to follow this up in the next PR since this is getting big

Reply:
actually, this is not explicitly defined in the product spec so will do this at a later time